openbb_terminal/helper_funcs.py

```diff
@@ -1612,52 +1561,70 @@ def check_list_values_from_valid_values_list(given_values: str) -> List[str]:
     return check_list_values_from_valid_values_list
 
 
-def get_preferred_source(command_path):
+def get_preferred_source(command_path: str):
     """
     Returns the preferred source for the given command. If a value is not available for the specific
     command, returns the most specific source, eventually returning the overall default source.
 
     Parameters
     ----------
-    command_path - The command to find the source for. Example would be "stocks.load" to return the value
-    for stocks.load first, then stocks, then the default value.
+    command_path: str
+        The command to find the source for. Example would be "stocks/load" to return the value
+        for stocks.load first, then stocks, then the default value.
 
     Returns
     -------
-    The preferred source for the given command
+    str:
+        The preferred source for the given command
     """
-
     try:
         with open(obbff.PREFERRED_DATA_SOURCE_FILE) as f:
             # Load the file as a JSON document
             json_doc = json.load(f)
 
             # We are going to iterate through each command as if it is broken up by period characters (.)
-            path_objects = command_path.split(".")
+            path_objects = command_path.split("/")[1:]
 
             # Start iterating through the top-level JSON doc to start
             deepest_level = json_doc
 
             # If we still have entries in path_objects, continue to go deeper
             while len(path_objects) > 0:
-
                 # Is this path object in the JSON doc? If so, go into that for our next iteration.
                 if path_objects[0] in deepest_level:
                     # We found the element, so go one level deeper
                     deepest_level = deepest_level[path_objects[0]]
+
                 else:
-                    # We didn't find the next level, so use the default value at this level
-                    return deepest_level["default"]
+                    # If we have not find the `load` on the deepest level it means we may be in a sub-menu
+                    # and we can use the load from the Base class
+                    if path_objects[0] == "load":
+
+                        # Get the context associated with the sub-menu (e.g. stocks, crypto, ...)
+                        context = command_path.split("/")[1]
+
+                        # Grab the load source from that context if it exists, otherwise throws an error
+                        if context in json_doc:
+                            if "load" in json_doc[context]:
+                                return json_doc[context]["load"]
+
+                    # We didn't find the next level, so flag that that command default source is missing
+                    # We decided to have these mentioned explicitly
+                    console.print(
+                        f"[red]'sources.json.default' file does not contain {command_path}[/red]"
```

Comment:
```suggestion
                        f"[red]'data_sources_default.json' file does not contain {command_path}[/red]"
```
Update older file name